app/forms/delivery_partner_form.rb

```diff
@@ -74,8 +73,8 @@ def lead_providers_and_cohorts_validation
 
     # Ensure all selected lead providers have at least one selected cohort
     # This is indicated by the presence of a provider relationship for that lead provider
-    lead_providers.each do |lead_provider|
-      unless provider_relationships.filter { |provider_relationship| provider_relationship.lead_provider == lead_provider }.any?
+    lead_providers.each do |lead_provider_id|
+      unless provider_relationships.filter { |provider_relationship| provider_relationship.lead_provider_id == lead_provider_id }.any?
```

Comment:
Possibly use pluck i.e getting all provider ids and checking the array if it includes lead_provider id

Reply:
Good shout